### PR TITLE
feat(eslint-plugin): add no-slot-attribute rule (VIV-1700)

### PR DESCRIPTION
**Pull request by RichardHelm**

### CODE REVIEW

File: libs/eslint-plugin/src/rules/no-slot-attribute.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import * as utils from 'eslint-plugin-vue/lib/utils';
+import type { Rule } from 'eslint';
+import { ComponentMetadata } from '../utils/ComponentMetadata';
+import { normalizeTag } from '../utils/components';
+
+const componentsWithNamedSlots = new ComponentMetadata<null>();
```
Comment (rachelbt) [inline libs/eslint-plugin/src/rules/no-slot-attribute.ts:6]: can't we do this automatically?
I'm afraid that we will miss component - like - banner and dial-pad are missing,

Comment (RichardHelm) [reply libs/eslint-plugin/src/rules/no-slot-attribute.ts:6]: You're right, we should get them automatically. But currently we have no way of doing it

I've checked the list again and updated it. Dial-pad doesn't have named slots so it's not included

Comment (rachelbt) [reply libs/eslint-plugin/src/rules/no-slot-attribute.ts:6]: can't see button (icon) + banner (icon + Action Items)

Comment (RichardHelm) [reply libs/eslint-plugin/src/rules/no-slot-attribute.ts:6]: You're right 🤦‍♂️ Updated

Comment (rachelbt) [reply libs/eslint-plugin/src/rules/no-slot-attribute.ts:6]: we need to find a better way for that, or maybe add all the component all together, as you are not specifying slots name. WDYT?

Comment (RichardHelm) [reply libs/eslint-plugin/src/rules/no-slot-attribute.ts:6]: I see your point but we currently don't have a way to get any metadata, not even a list of all components

**Review by rachelbt — APPROVED** [review]

**Review by rachelbt — APPROVED** [review]

**Pull request merged**